### PR TITLE
Fix Emulator Error & Add Flake Tracker to nightly notification

### PR DESCRIPTION
**Pull request by sunmou99**
1. Assert class has been moved from `junit.framework` to `org.junit.Assert`
Add `LargeTest` annotation. https://developer.android.com/reference/androidx/test/filters/LargeTest
2. For nightly test, if some tests failed, there will be a Flake Tracker

e.g.
### ❌&nbsp; Integration test FAILED
Requested by @DellaBitta on commit bff3d7aab2e6c9e40f58ab4f8127bf83784ba374
Last updated: Wed Jul 21 04:35 PDT 2021 
**[View integration test log & download artifacts](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1051911050)**
| Failures | Configs |
|----------|---------|
| missing_log | [BUILD] [ERROR] [Windows] [boringssl, openssl]<br/>[TEST] [ERROR] [Android] [macos] [emulator_32bit]<br/> |
| auth | [TEST] [ERROR] [Android] [ubuntu] [emulator_32bit]<br/>[TEST] [ERROR] [MacOS] [boringssl, openssl]<br/> |
| firestore | [TEST] [FAILURE] [Android] [ubuntu] [android_latest]<details><summary>(1 failed tests)</summary>&nbsp;&nbsp;TestDocumentSnapshotErrorReporting</details> |

Add flaky tests to **[go/fpl-cpp-flake-tracker](http://go/fpl-cpp-flake-tracker)**

### CODE REVIEW

File: scripts/gha/it_workflow.py [hunk 1]
```diff
@@ -62,6 +62,8 @@
 _COMMENT_TITLE_FAIL = "### ❌&nbsp; Integration test FAILED\n"
 _COMMENT_TITLE_SUCCEED = "### ✅&nbsp; Integration test succeeded!\n"
 
+_COMMENT_FLAKY_TRACKER = "Add flaky tests to **[Flake Tracker](https://docs.google.com/spreadsheets/d/1KQdg062mY2vTGq2w8bQ15CXopuXKQ2jro3Eob8hBnLQ/edit?usp=sharing&resourcekey=0-cjzOgusCPXYDVKLc-zyLOA)**\n"
```
Comment (DellaBitta) [inline scripts/gha/it_workflow.py:65]: Suggest making this the go link instead, for obfuscation purposes.  go/fpl-cpp-flake-tracker

**Review by DellaBitta — CHANGES_REQUESTED** [review]

File: scripts/gha/it_workflow.py [hunk 2]
```diff
@@ -62,6 +62,8 @@
 _COMMENT_TITLE_FAIL = "### ❌&nbsp; Integration test FAILED\n"
 _COMMENT_TITLE_SUCCEED = "### ✅&nbsp; Integration test succeeded!\n"
 
+_COMMENT_FLAKY_TRACKER = "Add flaky tests to **[go/fpl-cpp-flake-tracker](https://docs.google.com/spreadsheets/d/1KQdg062mY2vTGq2w8bQ15CXopuXKQ2jro3Eob8hBnLQ/edit?usp=sharing&resourcekey=0-cjzOgusCPXYDVKLc-zyLOA)**\n"
```
Comment (DellaBitta) [inline scripts/gha/it_workflow.py:65]: The URL in the parenthesis should be `http1://go/fpl-cpp-flake-tracker`

**Review by DellaBitta — CHANGES_REQUESTED** [review]

File: scripts/gha/it_workflow.py (hunk 2 above)
Comment (sunmou99) [reply scripts/gha/it_workflow.py]: fixed. And I believe you mean `http://go/fpl-cpp-flake-tracker`

Comment (DellaBitta) [reply scripts/gha/it_workflow.py]: Ah yes, that I did!

**Review by DellaBitta — APPROVED** [review]

**Comment by jonsimantov** [comment]
nit: I think I'd like the "Add flaky tests to go/fpl-cpp-flake-tracker" to be at the bottom, after the table.

**Comment by sunmou99** [comment]
> nit: I think I'd like the "Add flaky tests to go/fpl-cpp-flake-tracker" to be at the bottom, after the table.
Updated

File: scripts/gha/integration_testing/test_validation.py [hunk 3]
```diff
@@ -233,8 +233,8 @@ def summarize_test_results(tests, platform, summary_dir, file_name="summary.log"
       summary_json["failures"][testapp]["raw_result_link"] = test.raw_result_link
     failed_tests = re.findall(r"\[  FAILED  \] (.+)[.](.+)", results.summary)
     for failed_test in failed_tests:
-      failed_test = failed_test[1]
-      pattern = fr'\[ RUN      \] (.+)[.]{failed_test}(.*?)\[  FAILED  \] (.+)[.]{failed_test}'
+      failed_test = failed_test[0] + "." + failed_test[1]
+      pattern = fr'\[ RUN      \] {failed_test}(.*?)\[  FAILED  \] {failed_test}'
```
Comment (jonsimantov) [inline scripts/gha/integration_testing/test_validation.py:237]: nit: would it better to match ` *RUN *` (space-star-RUN-space-star) in the regexp, and similar with FAILED?

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**